php/src/Google/Protobuf/Internal/GPBUtil.php

```diff
@@ -597,4 +609,65 @@ public static function hasJsonValue($msg)
                is_a($msg, "Google\Protobuf\StringValue") ||
                is_a($msg, "Google\Protobuf\BytesValue");
     }
+
+
+    private static $wrapperTypes = [
+        DoubleValue::class,
+        FloatValue::class,
+        Int64Value::class,
+        UInt64Value::class,
+        Int32Value::class,
+        UInt32Value::class,
+        BoolValue::class,
+        StringValue::class,
+        BytesValue::class,
+    ];
+
+    /**
+     * Determine if a fully qualified class name is a wrapper type.
+     *
+     * @param string $class the fully qualified class name
+     * @return bool true if this is a wrapper type
+     */
+    public static function isWrapperType($class)
+    {
+        return in_array($class, self::$wrapperTypes);
```

Comment:
Will this linear search be a conformance concern? How about add a ```checkWrapperMessage``` which wrap ```checkMessage``` so that we only pay cost for wrapper messages?